pandas/_libs/tslibs/period.pyx

```diff
@@ -2614,6 +2614,13 @@ class Period(_Period):
             raise ValueError(msg)
 
         if ordinal is None:
+            if dt.tzinfo:
+                # GH 47005
+                warnings.warn(
+                    "The pandas.Period class does not support timezones. "
+                    "The timezone given in '%s' will be ignored." % value,
```

Comment:
Nit: Could you use an f-string here instead?

Reply:
Yap. My memory of how Python works and some post on StackOverflow suggested that %-formatting would be faster which I think is an important consideration here. But some quick testing with IPython's `%timeit` suggests it's the other way around at least on my system. I'll change this to an f-string in a couple hours.